org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -271,27 +272,34 @@ private TextEdit getOrganizeImportsProposal(IInvocationContext context, IProgres
 		}
 	}
 
-	private Optional<Either<Command, CodeAction>> getGetterSetterAction(CodeActionParams params, IInvocationContext context, IType type) {
+	private Optional<Either<Command, CodeAction>> getGetterSetterAction(CodeActionParams params, IInvocationContext context, IType type, String kind, IProgressMonitor monitor) {
 		try {
 			AccessorField[] accessors = GenerateGetterSetterOperation.getUnimplementedAccessors(type);
 			if (accessors == null || accessors.length == 0) {
 				return Optional.empty();
 			} else if (accessors.length == 1 || !preferenceManager.getClientPreferences().isAdvancedGenerateAccessorsSupported()) {
 				CodeActionProposal getAccessorsProposal = (pm) -> {
+					ASTNode declarationNode = null;
+					CompilationUnit astRoot = CoreASTProvider.getInstance().getAST(type.getCompilationUnit(), CoreASTProvider.WAIT_YES, monitor);
+					Range range = params.getRange();
+					if (astRoot != null && range != null) {
+						ASTNode node = NodeFinder.perform(astRoot, DiagnosticsHelper.getStartOffset(type.getCompilationUnit(), range), DiagnosticsHelper.getLength(type.getCompilationUnit(), range));
+						declarationNode = SourceAssistProcessor.getDeclarationNode(node);
+					}

```

Comment:
This block of code is used to check if the cursor is in a position of type declaration. Since the external invoker has already checked the cursor position, why not to pass it to `getGetterSetterAction` function directly?